### PR TITLE
Added Javascript fetch to website

### PR DESCRIPTION
**Pull request by dallen68**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,60 @@
...
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("comment", textComment);
+      commentEntity.setProperty("timestamp", timestamp);
+      
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+      response.sendRedirect("index.html");
   }
 }
+
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:68]: remove unused code, add newline to end of file

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,19 +14,60 @@
...
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Dante!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:37]: It's not immediately clear what the workflow from here on is doing, consider adding some comments.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,19 +14,60 @@
...
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> comments = new ArrayList<>();
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:42]: Best practice is to use the base interface List as common Collections utilities do not use Arraylist.
`List<String> comments = new ArrayList<>();`

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -14,19 +14,60 @@
...
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("comment");
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: I'm not too familiar with the library. Just making sure, is the cast here necessary? It seems that Entity has a toString() which should make the cast implicit when it's assigned to a String type.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -14,19 +14,60 @@
...
+      comments.add(comment);
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: nit: This and the line above could probably be 1 line without sacrificing clarity.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -14,19 +14,60 @@
...
+    }
+
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: nit: Could probably remove this extra newline

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (dallen68) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: I feel that it is easier for me to understand when they are seperate.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -27,24 +27,25 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.Lisy;
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: Typo in import

**Comment by dallen68** [comment]
Fixed typo

**Pull request merged**